src/NUnitFramework/framework/Constraints/CollectionOrderedConstraint.cs

```diff
@@ -297,6 +291,26 @@ private void CreateNextStep(string propertyName)
             _steps.Add(_activeStep);
         }
 
+        private object ExtractValue(IEnumerable actual, object item, string propertyName, int index)
+        {
+            object value;
+            PropertyInfo property = item.GetType().GetProperty(propertyName);
+            if (property == null)
+            {
+                FieldInfo field = item.GetType().GetField(propertyName);
+                if (field == null)
+                {
+                    throw new ArgumentException($"No property or field with name {propertyName} was found at index {index}", nameof(actual));
+                }
+                value = field.GetValue(item);
+            }
+            else
+            {
+                value = property.GetValue(item, null);
```

Comment:
Can you flip this around so that it's nearer to the `GetProperty` and `property == null` lines, to keep related things close together? Same for the `field.GetValue` line?